docs/workflow.md

```diff
@@ -0,0 +1,149 @@
+<!--
+TODO: should we call "this project" Juno? I saw in communication.md we did,
+but I'm not sure
+ -->
+
+# Workflow
+
+One of the goals of this project is to help users adopt a more efficient
+workflow when editing Julia code. To understand how to gain this productivity
+boost, we will consider a common workflow for many Julia developers and a Juno
+based workflow.
+
+## Status quo
+
+Suppose you are working on a file `foo.jl` that contains the definition of a
+type `Bar`. As you are hacking away on the fields and functionality of the `Bar`
+type you want to experiment with your work in the REPL. Here's what you might do:
+
+- Edit `foo.jl`
+- Fire up Julia and `include("foo.jl")` to load up your new types and methods
+- Return to the editor to make more changes to `foo.jl`
+- Go back to your REPL session and `include("foo.jl")` again, only to be greeted by the
+  following error message:
+
+```
```

Comment:
I didn't do it here because I wasn't displaying Julia source code... just REPL output.

I'm happy to try it if you think it would be better.